pkgs/java_http/lib/src/java_client.dart

```diff
@@ -230,24 +231,39 @@ class JavaClient extends BaseClient {
     return contentLength;
   }
 
-  void _responseBody(
+  Future<void> _responseBody(
     Uri requestUrl,
     HttpURLConnection httpUrlConnection,
     SendPort sendPort,
     int? expectedBodyLength,
-  ) {
+  ) async {
     final responseCode = httpUrlConnection.getResponseCode();
 
     final inputStream = (responseCode >= 200 && responseCode <= 299)
         ? httpUrlConnection.getInputStream()
         : httpUrlConnection.getErrorStream();
+    final bufferedInputStream = BufferedInputStream(inputStream);
+    print(bufferedInputStream.available());
 
-    int byte;
+    int bytesRead;
     var actualBodyLength = 0;
-    // TODO: inputStream.read() could throw IOException.
-    while ((byte = inputStream.read()) != -1) {
-      sendPort.send([byte]);
-      actualBodyLength++;
+    final buffer = JArray(jbyte.type, 4096);
+    // TODO: bufferedInputStream.read() could throw IOException.
+    while ((bytesRead = bufferedInputStream.read1(buffer, 0, buffer.length)) !=
+        -1) {
```

Comment:
I'm not too keen on the while loop condition going over to the next line, I think I will refactor it to make the code cleaner.

Reply:
:heavy_check_mark: Done!

To simplify the while loop condition, we no longer do the variable assignment within the loop condition.